src/main/java/org/wso2/carbon/connector/googlefirebase/FirebaseConfig.java

```diff
@@ -80,19 +80,27 @@ public void connect(MessageContext messageContext) {
                     log.debug("Started to initialize the firebase admin sdk with the provided firebase service " +
                             "account's credentials");
                 }
+                log.info("Started to initialize the firebase admin sdk with the provided firebase service " +
+                        "account's credentials");
                 FirebaseOptions options = FirebaseOptions.builder()
                         .setDatabaseUrl(FirebaseConstants.HTTPS + credentials.get(FirebaseConstants.API_PROJECT_ID)
                                 + FirebaseConstants.FIREBASE_IO)
                         .setCredentials(GoogleCredentials.fromStream(new ByteArrayInputStream(
-                                credentials.toString().getBytes())))
+                                credentials.toString().replaceAll("\\\\n", "\\n").getBytes())))
                         .build();
+                if(!FirebaseApp.getApps().isEmpty()) {
+                    FirebaseApp existingApp = FirebaseApp.getApps().get(0);
```

Comment:
`FirebaseApp.getApps().clear()` does not work as it is an immutable list.

Reply:
Why we are deleting only the first one , any reason ?

Reply:
The SDK allows to create multiple. We need only one app to work. When an app is added, it is added under the name "default". We just need to delete that.

Reply:
Shall we add that as a comment may be , would be helpful in future

Reply:
+1. Added.